openstack/client.go

```diff
@@ -321,5 +319,7 @@ func NewImageServiceV2(client *gophercloud.ProviderClient, eo gophercloud.Endpoi
 	if err != nil {
 		return nil, err
 	}
-	return &gophercloud.ServiceClient{ProviderClient: client, Endpoint: url}, nil
+	return &gophercloud.ServiceClient{ProviderClient: client,
+		Endpoint:     url,
+		ResourceBase: url + "v2/"}, nil
```

Comment:
The thing is, the endpoint is usually not created with "v2/" included.
Since gophercloud is intended to work with any openstack deployment it needs to be compatible with all the openstack deployments out there, otherwise, you need to make all of them update their endpoints

http://docs.openstack.org/mitaka/install-guide-ubuntu/glance-install.html